app/controllers/kudos_controller.rb

```diff
@@ -46,6 +47,11 @@ def destroy
     redirect_to kudos_url, notice: 'Kudo was successfully destroyed.'
   end
 
+  def correct_employee
+    @kudo = current_employee.given_kudos.find_by(id: params[:id])
```

Comment:
you already has a `@kudo` from the previous callback, overwriting it might lead to some comfusion later on.
 I would suggest something like:
`redirect_to kudos_path unless @kudo.giver_of_kudo == current_employee`

Reply:
also good point :), fixing